src/Access.php

```diff
@@ -224,7 +224,12 @@ public function get_post_date_to_access( $post_id = '' ) {
 			$post_id = $post->ID;
 		}
 
-		return get_post_meta( $post_id, SIMPLESHOP_PREFIX . 'date_to_access', true );
+		$date = get_post_meta( $post_id, SIMPLESHOP_PREFIX . 'date_to_access', true );
+		// Backwards compatibility.
+		if (strpos($date, '-') !== false) {
+			return $date;
+		}
+		return date('Y-m-d H:i:s', $date);
```

Comment:
Prosím o úpravu - není zřejmé, co se tady může vracet z `get_post_meta()`. za půl roku nebudeme vědět, co to dělá a proč to tu je. Buď upravte kód tak, aby testoval vysloveně špatný stav hodnoty (třeba nějaký konkrétní tvar) a nebo alespoň do komentáře napište, kvůli čemu to tu je a jaké nevalidní hodnoty tudy mohou přitékat.

A ještě, pokud je hodnota nevalidní (rspt. něco jiného než datum), jaký má smysl předávat tuto informaci rovnou do návratové hodnoty? Tím se potenciální problém přesouvá dál do aplikace. Když tu funkce má roli getteru, tak by mohla na sebe převzít funkci validátoru a případné neočekávané hodnoty buď opravit do požadovaného tvaru a nebo pak předat objektivně prázdnou hodnotu (`null` a nebo prázdný string `''`). Prostě minimalizovat situaci, že aplikací protéká bordel, který aplikace nechápe.